tensorflow_datasets/image/fruits360.py

```diff
@@ -0,0 +1,125 @@
+# coding=utf-8
+# Copyright 2019 The TensorFlow Datasets Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Fruits-360: A dataset of images containing fruits.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+
+from tensorflow.python.framework import errors
+
+import tensorflow_datasets.public_api as tfds
+from tensorflow import io as tfio
+
+_CITATION = """\
+@article{article,
+author = {Mureșan, Horea and Oltean, Mihai},
+year = {2018},
+month = {06},
+pages = {26-42},
+title = {Fruit recognition from images using deep learning},
+volume = {10},
+journal = {Acta Universitatis Sapientiae, Informatica},
+doi = {10.2478/ausi-2018-0002}
+}
+"""
+
+_COMMIT_SHA = 'b6960fc0287be349c222c3dbdd2624958c902431'
+_DOWNLOAD_URL = 'https://github.com/Horea94/Fruit-Images-Dataset/archive/{sha}.tar.gz'.format(sha=_COMMIT_SHA)
+
+_IMAGE_SIZE = 100
+_IMAGE_SHAPE = (_IMAGE_SIZE, _IMAGE_SIZE, 3)
+_CLASS_NAMES = ['Apple Braeburn', 'Apple Crimson Snow', 'Apple Golden 1', 'Apple Golden 2', 'Apple Golden 3',
```

Comment:
Dataset updates are monthly, so you should not type static class names. If it must be dynamic, you can get the class names from the dataset folder names.

Reply:
Label IDs are deterministic already without sorting, since the pair (class_name, filename) uniquely identifies an image.